packages/kit-bg/src/states/jotai/jotaiStorage.ts

```
@@ -102,33 +106,124 @@ export function atomWithStorage<Value>(
+      get,
+      set,
+      update: IJotaiSetStateActionWithReset<Value | Promise<Value>>,
+    ) => {
+      jotaiVerify.ensureNotPromise(update);
+      const p = createPromiseTarget<boolean>();
+
+      set(baseAtom, async (prevValue) => {
+        const value = (async () => {
+          if (prevValue instanceof Promise) {
+            // eslint-disable-next-line no-param-reassign
+            prevValue = await prevValue;
```

Biome (1.9.4)

[error] Reassigning a function parameter is confusing.

The parameter is declared here:

Use a local variable instead.

(lint/style/noParameterAssign)